packages/tutorial/src/tools.tsx

```diff
@@ -0,0 +1,64 @@
+import * as AI from 'ai-jsx';
+import { UseTools } from 'ai-jsx/batteries/use-tools';
+import z from 'zod';
+import yahooFinance from 'yahoo-finance2';
+import asciichart from 'asciichart';
+import enquirer from 'enquirer';
+
+const { prompt } = enquirer;
+
+function StockAgent(props: { query: string }) {
+  async function checkStockPrice(symbol: string) {
+    const quote = await yahooFinance.quote(symbol);
+    return quote.regularMarketPrice ?? 'Unknown';
```

Comment:
I kept getting an error here when I didn't use a proper symbol. You could check if `quote` is undefined or not.